### PR TITLE
FSA22V2-216: Add support for dependabot

### PR DESCRIPTION
**Pull request by marissahuysentruyt**
### Description:

Add dependabot.yml file to check for weekly npm updates to dependencies. This file is in the `.github` folder, named `dependabot.yml`. 

### Spec:

See Story: [FSA22V2-216](https://sparkbox.atlassian.net/browse/FSA22V2-216)

### Validation:

* [x] This PR has code changes, and our linters still pass.

#### To Validate:

1. Make sure all PR Checks have passed (GitHub Actions, CircleCI, Code Climate, etc).
2. Pull down all related branches.
3. Confirm `dependabot.yml` file: 
    a. is valid
    b. is named correctly
    c. lives in the correct place in this repo

---

### Pending Questions
1. As Quinn mentioned in a comment, should we add an `open-pull-request-limit` to the dependabot file at all? What would the limit be? 
2. Similarly, should we be more specific about the schedule (day and/or time) that the update checks run? Is there a better or worse day? 
3. Since Sparkbox has specific branch naming strategies, should we utilize `pull-request-branch-name.separator` to make sure those created branches are consistent?
4. Quinn mentioned potentially needing to add `allow: - dependency-name: "react*` because React is a dependency. Is this already done by default?

### CODE REVIEW

**Review by bsmrdel101 — DISMISSED** [review]
LGTM!

**Review by QuinnrElder — COMMENTED** [review]
This looks great! If we update the version piece I can approve; great job!

Food for thought ~ After looking over the documentation for `dependabot.yml files` I noticed some things that we may want to take a look at implementing in the future...

If you look at these comments below and agree; maybe bring them up at stand-up or over a 1:1 with an instructor. It is never a bad idea to show room for improvement with a current approach or ticket. 

- Because the `dependabot.yml file` will auto-generate PRs for us. We may want to take a look at implementing the:
    - ```open-pull-requests-limit``` ~ to limit the amount it can have in the backlog
    - ```schedule.time & schedule.day``` ~ so we can specify when this is run and updated
    - ```pull-request-branch-name.separator``` ~ Sparkbox has specific branching strategies that we like to use so we might need the Branches created from this script to match the existing patterns.
    - [allow](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#allow) ~ because React is a dependency of this if not by default add ```allow: - dependency-name: "react*"```
    
Again just talking out loud here...

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Basic dependabot.yml file with
+# minimum configuration for npm
+
+version: 2
```
[the remark this answers is not in the document]
Comment (dustin-jw) [reply .github/dependabot.yml:4]: This `version` means something different–it's not our project's version. It's related to Dependabot itself, I think. I don't see any documentation for it, but all the code samples have it set to `2`. This looks good to me as-is!

**Review by dustin-jw — DISMISSED** [review]

File: .github/dependabot.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (QuinnrElder) [reply .github/dependabot.yml:4]: oops ~ good call! My bad on that one!

**Review by dustin-jw — APPROVED** [review]

**Review by bsmrdel101 — APPROVED** [review]
Still LGTM!

**Pull request merged**